prow/githubeventserver/githubeventserver.go

```diff
@@ -90,11 +89,9 @@ func (g *GitHubEventServer) GetHTTPServeMux() *http.ServeMux {
 	return g.httpServeMux
 }
 
-// Listen runs an http server by using the existing http.ServeMux, port and the
-// given grace period duration
-func (g *GitHubEventServer) Listen(gracePeriod time.Duration) {
-	defer interrupts.WaitForGracefulShutdown()
-	interrupts.ListenAndServe(&http.Server{Addr: ":" + strconv.Itoa(g.port), Handler: g.httpServeMux}, gracePeriod)
+// GetHTTPServer returns an http server using the github event server's port and http serve mux handler
+func (g *GitHubEventServer) GetHTTPServer() *http.Server {
```

Comment:
`interrupts.ListenAndServe()` just expects an interface -- can we make `GitHubEventServer` implement that interface, so no function is necessary to expose the internal server?

Reply:
Thanks @stevekuznetsov. I should have done this from the beginning. Addressed